maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/booterclient/JarManifestForkConfigurationTest.java

```diff
@@ -0,0 +1,190 @@
+        when( JarManifestForkConfiguration.toClasspathElementUri( anyString(), anyString(), any( File.class ) ) )
+                .thenCallRealMethod();
+        File dump = new File( BASE_DIR, "target/test-dump" );
+        assertThat( JarManifestForkConfiguration.toClasspathElementUri( parent, classPathElement, dump ) )
+                .isEqualTo( "../../../Test%20User/me/.m2/repository/grp/art/1.0/art-1.0.jar" );
+    }
+
+    @Test
+    public void crossDriveWindows()
+        throws Exception
+    {
+        mockStatic( JarManifestForkConfiguration.class );
+        mockStatic( InPluginProcessDumpSingleton.class );
+        when( InPluginProcessDumpSingleton.getSingleton() ).thenReturn(mock( InPluginProcessDumpSingleton.class ));
+        String parent = "C:\\Windows\\Temp\\surefire";
+        String classPathElement = "X:\\Users\\me\\.m2\\repository\\grp\\art\\1.0\\art-1.0.jar";
+        when( JarManifestForkConfiguration.relativize( parent, classPathElement ) )
+                .thenThrow( new IllegalArgumentException() );
+        when( JarManifestForkConfiguration.toAbsoluteUri( classPathElement ) )
+                .thenReturn( "file:///X:/Users/me/.m2/repository/grp/art/1.0/art-1.0.jar" );
+        when( JarManifestForkConfiguration.toClasspathElementUri( anyString(), anyString(), any( File.class ) ) )
+                .thenCallRealMethod();
+        File dump = new File( BASE_DIR, "target/test-dump" );
+        assertThat( JarManifestForkConfiguration.toClasspathElementUri( parent, classPathElement, dump ) )
+                .isEqualTo( "file:///X:/Users/me/.m2/repository/grp/art/1.0/art-1.0.jar" );
+    }
+
+    @Test
+    public void shouldRelativizeOnRealPlatform()
+            throws Exception
+    {
+        File parentDir = new File( BASE_DIR, "target/test-parent" );
+        //noinspection ResultOfMethodCallIgnored
+        parentDir.mkdir();
+        File testDir = new File( BASE_DIR, "target/test with white spaces" );
+        //noinspection ResultOfMethodCallIgnored
+        testDir.mkdir();
+
+        String relativeTestDir =
+                JarManifestForkConfiguration.relativize( parentDir.getCanonicalPath(), testDir.getCanonicalPath() );
+
+        assertThat( relativeTestDir )
+                .isEqualTo( ".." + File.separator + "test with white spaces" );
+    }
+
+    @Test
+    public void shouldMakeAbsoluteUriOnRealPlatform()
+            throws Exception
+    {
+        File testDir = new File( BASE_DIR, "target/@2 test with white spaces" );
+        //noinspection ResultOfMethodCallIgnored
+        testDir.mkdir();
+
+        URI testDirUri = new URI( JarManifestForkConfiguration.toAbsoluteUri( testDir.getCanonicalPath() ) );
+
+        assertThat( testDirUri.getScheme() )
+                .isEqualTo( "file" );
+
+        assertThat( testDirUri.getRawPath() )
+                .isEqualTo( testDir.toURI().getRawPath() );
```

Comment:
This will likely fail when run on Windows using a project directory that is a UNC path rather than a drive letter: just as `File.toURL` had numerous problems corrected only by `File.toUri`, so `File.toUri` has some problems corrected by `Path.toUri`.

I would suggest dropping `shouldRelativizeOnRealPlatform` and `shouldMakeAbsoluteUriOnRealPlatform`. If you want unit tests, that is what I originally wrote. If you want something more realistic, demonstrating that the _result_ is what was intended, `shouldMakeRelativeUriOnRealPlatform` is better.

Reply:
> I am wondering who wants to run Jenkins build on Surefire project that is located on shared network folder

`\\vboxsvr\linux` perhaps?